Managed Identity support in Azure Kusto output plugin (#1578)
fluent/fluent-bit-docs: pipeline/outputs/azure_kusto.md

| Key                         | Description                                                                                                                                                                                                                      | Default     |
| --------------------------- | -------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | ----------- |

> | tenant_id                   | _Only one of AAD registered application or Managed Identity may be set_ - The tenant/domain ID of the AAD registered application.                                                                                                                                                             |             |

Reviewer: hi @dceravigupta! thanks for submitting a docs PR 🙂

I want to offer a few edits for clarity and consistency, and one thing I want to ask about here is the "Only one of AAD registered application or Managed Identity may be set" note: does "AAD registered application" correspond to the `tenant_id`, `client_id`, and `client_secret` parameters? and does "Managed Identity" correspond to `managed_identity_client_id`?

it sounds like you MUST use either `tenant_id` + `client_id` + `client_secret` in combination, OR you can use `managed_identity_client_id`, but I want to make sure I understand things correctly before making further suggested edits.

Author: Yes,  AAD registered application correspond to the tenant_id, client_id, and client_secret parameters. And "Managed Identity" correspond to managed_identity_client_id.

`it sounds like you MUST use either tenant_id + client_id + client_secret in combination, OR you can use managed_identity_client_id, but I want to make sure I understand things correctly before making further suggested edits.`
Correct

> | tenant_id                   | _Only one of AAD registered application or Managed Identity may be set_ - The tenant/domain ID of the AAD registered application.                                                                                                                                                             |             |

Reviewer: ```suggestion
| tenant_id                   | _Required if `managed_identity_client_id` is not set_ - The tenant/domain ID of the AAD registered application.                                                                                                                                                             |             |
```

> | client_id                   | _Only one of AAD registered application or Managed Identity may be set_ - The client ID of the AAD registered application.                                                                                                                                                                    |             |

Reviewer: ```suggestion
| client_id                   | _Required if `managed_identity_client_id` is not set_ - The client ID of the AAD registered application.                                                                                                                                                                    |             |
```

> | client_secret               | _Only one of AAD registered application or Managed Identity may be set_ - The client secret of the AAD registered application ([App Secret](https://docs.microsoft.com/en-us/azure/active-directory/develop/howto-create-service-principal-portal#option-2-create-a-new-application-secret)). |

Reviewer: ```suggestion
| client_secret               | _Required if `managed_identity_client_id` is not set_ - The client secret of the AAD registered application ([App Secret](https://docs.microsoft.com/en-us/azure/active-directory/develop/howto-create-service-principal-portal#option-2-create-a-new-application-secret)). |
```

> | managed_identity_client_id  | _Only one of AAD registered application or Managed Identity may be set_ - The managed identity id to authenticate with. Set to "SYSTEM" for system-assigned managed identity. Set the MI client ID (GUID) for user-assigned managed identity.                                                                                                                                                                   |             |

Reviewer: ```suggestion
| managed_identity_client_id  | _Required if `tenant_id`, `client_id`, and `client_secret` are not set_ - The managed identity ID to authenticate with. Set to `SYSTEM` for system-assigned managed identity, or set to the MI client ID (GUID) for user-assigned managed identity.                                                                                                                                                                   |             |
```
